apps/test/unit/templates/certificates/PetitionCallToActionTest.js

```diff
@@ -1,34 +1,81 @@
 import React from 'react';
 import {isolateComponent} from 'isolate-react';
 import {expect} from '../../../util/reconfiguredChai';
+import sinon from 'sinon';
+import $ from 'jquery';
+
+const isolateCallToAction = props =>
+  isolateComponent(
+    <PetitionCallToAction gaPagePath={'/congrats/coursetest-2030'} {...props} />
+  );
 
 describe('PetitionCallToAction', () => {
   it('has a petition message', () => {
-    const callToAction = isolateComponent(<PetitionCallToAction />);
     expect(
-      callToAction.findOne('#petition-message').content().length
+      isolateCallToAction()
+        .findOne('#petition-message')
+        .content().length
     ).to.be.greaterThan(0);
   });
   it('has a message to sign the petition', () => {
-    const callToAction = isolateComponent(<PetitionCallToAction />);
     expect(
-      callToAction.findOne('#sign-message').content().length
+      isolateCallToAction()
+        .findOne('#sign-message')
+        .content().length
     ).to.be.greaterThan(0);
   });
   it('has a form with submit button', () => {
-    const callToActionWithForm = isolateComponent(<PetitionCallToAction />);
+    const callToActionWithForm = isolateCallToAction();
     callToActionWithForm.inline(PetitionForm);
     const form = callToActionWithForm.findOne('form');
     expect(form.findOne('Button').content().length).to.be.greaterThan(0);
   });
-  it('has a form with five fields ', () => {
-    // TODO: Once the request is being made,
-    //  refactor this to instead check that the request is sending the correct data
-    const callToActionWithForm = isolateComponent(<PetitionCallToAction />);
+  it('does not send request if there are invalid fields', () => {
+    const requestSpy = sinon.spy($, 'ajax');
+
+    const callToActionWithForm = isolateCallToAction();
+    callToActionWithForm.inline(PetitionForm);
+    const form = callToActionWithForm.findOne('form');
+    form.props.onSubmit({preventDefault: sinon.stub()});
+
+    expect(requestSpy).to.have.been.notCalled;
+    $.ajax.restore();
+  });
+  it('sends request if there are no invalid fields', () => {
+    const requestSpy = sinon.spy($, 'ajax');
+    sinon.stub();
+
+    const callToActionWithForm = isolateCallToAction();
     callToActionWithForm.inline(PetitionForm);
-    const fields = callToActionWithForm.findAll('ControlledFieldGroup');
-    expect(fields).to.have.length(5);
+    // callToActionWithForm.inline(ControlledFieldGroup);
+    // ^^ If I do this, I get that fields[0].props is empty, so I can't do the props.onChange
+    const form = callToActionWithForm.findOne('form');
+
+    const fields = callToActionWithForm.findAll(ControlledFieldGroup);
+    console.log('form is', form);
+    console.log(fields.length, fields[0].props);
+    fields[0].props.onChange({
+      persist: sinon.stub(),
+      target: {name: 'name', value: 'fake name'}
+    });
+    form.props.onSubmit({preventDefault: sinon.stub()});
```

Comment:
I'm submitting these because it console logs the data –– it's clear the data isn't being updating here. I think it's because I don't have `.inline(ControlledFieldGroup)` so the components can't talk to each other (see https://davidmfoley.github.io/isolate-react/docs/isolateComponent/test-multiple-components). But I can't get the onChange otherwise